ddpui/api/charts_api.py

```
@@ -0,0 +1,1010 @@
+
+    # Get org warehouse
+    org_warehouse = OrgWarehouse.objects.filter(org=orguser.org).first()
+    if not org_warehouse:
+        raise HttpError(404, "Warehouse not configured")
+
+    # Use the common function to generate data and config
+    try:
+        result = generate_chart_data_and_config(payload, org_warehouse)
+        return ChartDataResponse(data=result["data"], echarts_config=result["echarts_config"])
+    except ValueError as e:
+        raise HttpError(400, str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)